### PR TITLE
Add changelog for 2024.07.0

### PR DESCRIPTION
**Pull request by yuvipanda**

### CODE REVIEW

File: docs/source/changelog.md [hunk 1]
```diff
@@ -1,5 +1,33 @@
...
+
+- add Python 3.12 [#1347](https://github.com/jupyterhub/repo2docker/pull/1347) ([@minrk](https://github.com/minrk))
+- [MRG] Update to JupyterLab 4 [#1340](https://github.com/jupyterhub/repo2docker/pull/1340) ([@jtpio](https://github.com/jtpio))
+- [MRG] Add CKAN content provider [#1336](https://github.com/jupyterhub/repo2docker/pull/1336) ([@u10313335](https://github.com/u10313335))
+
+### Bugs fixed
+
+- Docker engine: replace ContainerConfig with Config [#1355](https://github.com/jupyterhub/repo2docker/pull/1355) ([@manics](https://github.com/manics))
+- fix julia semver tilde upper bounds [#1351](https://github.com/jupyterhub/repo2docker/pull/1351) ([@minrk](https://github.com/minrk))
+- Remove support for deprecated Julia REQUIRE, Julia < 1.3 [#1350](https://github.com/jupyterhub/repo2docker/pull/1350) ([@minrk](https://github.com/minrk))
```
Comment (manics) [inline docs/source/changelog.md:17]:
```suggestion
([full changelog](https://github.com/jupyterhub/repo2docker/compare/2024.03.0...2024.06.0))

### Breaking changes

- [MRG] Update to JupyterLab 4 [#1340](https://github.com/jupyterhub/repo2docker/pull/1340) ([@jtpio](https://github.com/jtpio))
- Remove support for deprecated Julia REQUIRE, Julia < 1.3 [#1350](https://github.com/jupyterhub/repo2docker/pull/1350) ([@minrk](https://github.com/minrk))

### Enhancements made

- add Python 3.12 [#1347](https://github.com/jupyterhub/repo2docker/pull/1347) ([@minrk](https://github.com/minrk))
- [MRG] Add CKAN content provider [#1336](https://github.com/jupyterhub/repo2docker/pull/1336) ([@u10313335](https://github.com/u10313335))

### Bugs fixed

- Docker engine: replace ContainerConfig with Config [#1355](https://github.com/jupyterhub/repo2docker/pull/1355) ([@manics](https://github.com/manics))
- fix julia semver tilde upper bounds [#1351](https://github.com/jupyterhub/repo2docker/pull/1351) ([@minrk](https://github.com/minrk))
```

**Comment by yuvipanda** [comment]
Let's make this release on Monday, so that the JupyterLab 4 change doesn't cause breakage over the weekend

**Comment by manics** [comment]
If it's Monday it should be ` 2024.07.0` :smile:
There's no harm releasing before Monday, JupyterLab 4 was deployed to mybinder.org a while ago, so the breaking change only affects people installing rep2docker locally from pypi.

**Review by consideRatio — APPROVED** [review]

**Comment by yuvipanda** [comment]
Thanks @manics and @consideRatio.

**Pull request merged**